README.md

```diff
@@ -48,177 +83,849 @@ If you do not already have a **Developer Edition Account**, you can create one a
+- Multifactor authentication (MFA) - strengthens the security of password-based authentication by requiring additional verification of another factor such as a temporary one-time password or an SMS passcode. SDK supports user enrollment with MFA factors enabled by the administrator, as well as MFA challenges based on your Okta Sign-On Policy.
+- Unlock account - unlocks user account if it has been locked out due to excessive failed login attempts. **This functionality is subject to the security policy set by the administrator**.
+- Recover password - allows users to securely reset their password if they've forgotten it. **This functionality is subject to the security policy set by the administrator**
+- Restore authentication/unlock/recover transaction with the state token
+
+To initiate particular flow please make call to one of the available functions in `OktaAuthSdk.swift`:
 ```swift
-let client = AuthenticationClient(oktaDomain: URL(string: "https://{yourOktaDomain}")!, delegate: self)
+// Authentication
+public class func authenticate(with url: URL,
+                               username: String,
+                               password: String?,
+                               onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                               onError: @escaping (_ error: OktaError) -> Void)
+
+// Unlock account
+public class func unlockAccount(with url: URL,
+                                username: String,
+                                factorType: OktaRecoveryFactors,
+                                onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                                onError: @escaping (_ error: OktaError) -> Void)
+
+// Forgot password
+public class func recoverPassword(with url: URL,
+                                  username: String,
+                                  factorType: OktaRecoveryFactors,
+                                  onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                                  onError: @escaping (_ error: OktaError) -> Void)
+
+// Restore authentication
+public class func fetchStatus(with stateToken: String,
+                              using url: URL,
+                              onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                              onError: @escaping (_ error: OktaError) -> Void)
 ```
-[//]: # (end: createClient)
 
-The client must implement the [`AuthenticationClientDelegate`](AuthenticationClientDelegate) protocol. 
+Please note that closure `onStatusChange` returns `OktaAuthStatus` instance as a parameter. Instance of `OktaAuthStatus` class represents the current status that is returned by the server. It is developer's responsibilty to handle current status in order to proceed with the initiated flow. Check the status type by calling `status.statusType` and downcast to concrete status class. Example of  handling function could the following:
```

Comment:
Maybe a nit: does the closure return `OktaAuthStatus`, or is `OktaAuthStatus` provided to the closure?

Reply:
`OktaAuthStatus ` is returned by the closure via parameter. So this is classic approach where callback delivers data via parameter